doc/ug/dv_methodology.md

```diff
@@ -299,65 +299,124 @@ Once the regression tool is developed, it will provide a way to enable these cap
+  Could also be the intersection of two cover points.
+  One example could be observing the FIFO full signal and the FIFO write enable signal is asserted at the same time.
+  This is called a coverage cross and will give valuable information on whether this scenario was ever seen. 
+* **Cover Property coverage**: This observes events or series of events.
+  As an example think of the TL UL register bus used.
+  A cover property cover point could be the handshaking between valid and ready.
+
+
+### Why do we need coverage
+
+The common answer is to flush out bugs in the design.
+This is not accurate enough. Making sure there are no bugs in a design, while important it is not sufficient.
+One must also make sure the design work as intended.
+That is providing all the functionality specified in the design specification.
+So a more precise answer as to why we need coverage is to flush out flaws in the design.
+Flaws being both design bugs and deficiencies in the design with respect to the specification.
+
+#### When are we done testing?
+Another reason as to why we need coverage is to answer the seemingly simple but important question:
+**When are we done testing?**
+Do we need 1, 10, or 100 tests and should they run 10,100, or 1000 regressions?
+Only coverage can answer this question for you.
+
+#### Do we need both types of coverage
+
+Reaching a 100% code coverage does not tell you the whole story.
+Think of the simple example of an AND gate. 
+Input A, B, and output X. to get to a 100 percent code coverage only two different input combinations are needed.
+00 and 11, these two will produce all possible outcomes of X.
+
+![single AND gate](dv_method_single_gate.svg)
+
+
+The coverage will indicate that all code was exercised.
+But We do not know that our design work as intended. 
+All we know is that bot A, B, and X have been observed to take on both logic  0 and 1.
+We could not say for certain that the design was in fact an AND gate could just as easily be an OR gate.
+So we need functional coverage to tell us this.
+The first thing functional coverage will tell us is if we observed all possible values on the inputs.
+And by Adding a cross between the inputs and the output it will tell us which gate we are looking at.
+
+
+Reaching 100 percent functional coverage is not enough either.
+Remember functional coverage requires the designer to manually add coverage point into the design.
+Going back to our AND gate, let us say we take two of these and OR the outputs of the two.
+
+![multi gate](dv_method_multi_gate.svg)
+
+
+If we only add the cover points from our original example.
+These will still exercise the new output of our system and therefore result in reaching 100 percent functional coverage.
+But half of the design was not exercised.
+This is called a coverage hole and code coverage would have indicated that a part of the code was never exercised.
+
+So while Functional coverage will tell you if your design is working correctly.
```

Comment:
nit: Functional -> functional
functional coverage alone can't make sure design is working correctly.
I feel functional coverage tell you if the random stimulus works as you planned, while code coverage tell if there is some coverage hole in your test stimulus. both are needed.

Reply:
@weicaiyang  I corrected the uppercase error.
Not sure I understand the second part of the comment. Is that a reword suggestion?

Reply:
perhaps change `working correctly` to `working as planned`

Reply:
@weicaiyang I somehow feel "correctly" is more concise than planned which is a bit fluffy.
but that is my opinion - do you feel strongly about this?

Reply:
slightly feel if functional coverage already tell you design is working correctly, perhaps no need to have code coverage.
but I think it's minor. I'm fine to keep as is.

Reply:
@weicaiyang  functional coverage will tell you if the functions you are covering are working correctly.
Code coverage will tell you if you did indeed test it all.